app/jiraapi.py

```diff
@@ -281,13 +281,32 @@ def create_application_approval_task(self, application, domains, closing_date=No
             issuetype_name=issuetype_name,
             labels=[application.type] if application.type else []
         )
-
-        new_issue = self.generic_jira.create_issue(**details)
-        new_issue.update({
-            self.jira_field_codes['APPLICATION_FIELD_CODE']: str(application.id),
-            self.jira_field_codes['SUPPLIER_FIELD_CODE']: str(application.supplier_code)
-            if application.supplier_code else str(0)
-        })
+        existing_issues = self.generic_jira.jira.search_issues('"Marketplace Application ID" ~ "{}" '
+                                                               'AND issuetype = "{}"'
+                                                               .format(str(application.id), issuetype_name))
+        if len(existing_issues) > 0 and closing_date is None and application.type == 'new':
+            new_issue = existing_issues[0]
+            new_issue.update({'summary': summary,
+                              'duedate': pendulum.now().add(weeks=2).to_date_string(),
+                              self.jira_field_codes['SUPPLIER_FIELD_CODE']: str(application.supplier_code)
+                              if application.supplier_code else str(0)
+                              })
+            if new_issue.fields.status.name == 'Closed':
+                self.generic_jira.jira.transition_issue(new_issue, 'Reopen')
+        elif len(existing_issues) > 0 and closing_date is not None and application.type == 'new':
```

Comment:
Can you reduce some duplication by merging the first two if blocks?  Something like:
```
if len(existing_issues) > 0 and application.type == 'new':
    new_issue = existing_issues[0]
    if closing_date is None:
        new_issue.update(...)
            if new_issue.fields.status.name == 'Closed':
                ...
    else:
        new_issue.update(...)
```